### DISCUSSION TITLE
Node sends message content of 'NDEATH'

### DISCUSSION THREAD

**Discussion by dongzi114** [thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus#81] [General]
According to the 2.2 and 3.0 versions of the SparkLugB specification, there is no requirement to send "seq" information when sending "NDEATH" topics, especially in the 3.0 protocol version which emphasizes that “The NDEATH message MUST NOT include a
sequence number”. In fact, our nodes do include this content when sending "NDEATH".

**Comment by thomassorensen2650** [comment]
NDEATH should contain bdSeq and not seq.. are you seeing seq in NDEATH?

**Reply by dongzi114** [reply]
Yes. when my edge gateway loses power, it will send the following message to the "NDEATH" topic:

{
"timestamp" : 1727070986223,
"metrics" : [ {
"name" : "bdSeq",
"dataType" : "Int64",
"value" : 0
} ],
"seq" : 0
}

. Note that I captured the above text through MQTT.fx.

**Reply by dongzi114** [reply]
According to the content format of NDEATH in the SparkLugB 3.0 protocol:
{
"timestamp": 1486144502122,
"metrics": [{
"name": "bdSeq",
"timestamp": 1486144502122,
"dataType": "UInt64",
"value": 0
}]
}
Note that the data type in the specification is uint64, but in reality, node red sends int64. Also, the timestamp is reflected twice in the message, but in reality it was only sent once. I wonder if this will have any impact?

**Reply by dongzi114** [reply]
May I ask you a question? Are we developing based on the SparkLugB 2.2 specification? If the host application is developed based on the SparkLugB 3.0 specification, will there be any communication issues between the two? I once tried using VTSCADA and Ignition as host applications. When node red sends the following message to "NDEATH", it clears the data already received by the host. However, when I use ifix 2024 as the host application, IFIX does not clear the data already received by the host when node red sends the following message to "NDEATH".

{
"timestamp" : 1727070986223,
"metrics" : [ {
"name" : "bdSeq",
"dataType" : "Int64",
"value" : 0
} ],
"seq" : 0
}

Ifix 2024 uses a protocol based on SparkLugB 3.0, while Vtscada uses version 2.2. I haven't figured out which version of the protocol IGnition is using yet.

**Comment by thomassorensen2650** [comment]
Ahh, I can see the problem.. I will get that fixed... the Primary Application in the nodes listen to both Sparkplub B 2.2 and Sparkplug B 3 States, so it should work whatever format the host application use.

**Reply by dongzi114** [reply]
Thank you very much for your maintenance. We look forward to releasing a new version as soon as possible. Wishing you a pleasant work experience!

**Reply by dongzi114** [reply]
Dear Sir, I have been following the update of the version of "node red contrib mqtt spark plug plus" these days. May I ask if the new version can be released as soon as possible? Thank you very much!

**Comment by thomassorensen2650** [comment]
It's fixed and merged into the main branch, I'll see if I can release a new version this weekend.

**Reply by dongzi114** [reply]
Thank you very much!

**Comment by thomassorensen2650** [comment]
V2.1.11 Released with the fix
